File: lab1-js/lab_js_oop/square.js
import {Rectangle} from './rectangle.js'

export class Square extends Rectangle {
	constructor(side, color) {
		super(side, side, color)
		this.type = 'Квадрат'
	}

	repr() {
		return `${this.type} ${this.color.color} цвета с длиной стороны ${this.width}, площадью ${this.area()}.`
	}
}

File: lab1-test/lab_python_oop/square.py
from .rectangle import Rectangle


class Square(Rectangle):
    type = "Квадрат"

    def __init__(self, side, color):
        self.side = side
        super().__init__(self.side, self.side, color)

    def __repr__(self):
        return '{} {} цвета с длиной стороны {}, площадью {}.'.format(
            self.type,
            self.color.val,
            self.side,
            self.area()
        )

